src/functions/DidSiopMetadata.ts

```diff
@@ -1,12 +1,143 @@
+  SupportedTypes,
+} from '../types/SIOP.types';
 
-export function assertValidMetadata(opMetadata: DiscoveryMetadataPayload, rpMetadata: RPRegistrationMetadataPayload) {
-  verifyCredentialsFormat(rpMetadata.credential_formats_supported, opMetadata.credential_formats_supported);
+const allDidMethodsSupported = [
+  'did:3:',
+  'did:abt:',
+  'did:aergo:',
+  'did:ala:',
+  'did:amo:',
+  'did:bba:',
+  'did:bid:',
+  'did:bnb:',
+  'did:bryk:',
+  'did:btcr:',
+  'did:ccp:',
+  'did:celo:',
+  'did:com:',
+  'did:corda:',
+  'did:did:',
+  'did:dns:',
+  'did:dock:',
+  'did:dom:',
+  'did:dual:',
+  'did:echo:',
+  'did:elastos:',
+  'did:elem:',
+  'did:emtrust:',
+  'did:ens:',
+  'did:eosio:',
+  'did:evan:',
+  'did:example:',
+  'did:erc725:',
+  'did:etho:',
+  'did:ethr:',
+  'did:factom:',
+  'did:future:',
+  'did:gatc:',
+  'did:git:',
+  'did:github:',
+  'did:grg:',
+  'did:hedera:',
+  'did:holo:',
+  'did:hpass:',
+  'did:icon:',
+  'did:infra:',
+  'did:io:',
+  'did:ion:',
+  'did:iota:',
+  'did:ipid:',
+  'did:is:',
+  'did:iwt:',
+  'did:jlinc:',
+  'did:jnctn:',
+  'did:jolo:',
+  'did:keri:',
+  'did:key:',
+  'did:kilt:',
+  'did:klay:',
+  'did:kr:',
+  'did:life:',
+  'did:lit:',
+  'did:meme:',
+  'did:meta:',
+  'did:moac:',
+  'did:monid:',
+  'did:morpheus:',
+  'did:mydata:',
+  'did:near:',
+  'did:nft:',
+  'did:ockam:',
+  'did:onion:',
+  'did:ont:',
+  'did:omn:',
+  'did:op:',
+  'did:orb:',
+  'did:panacea:',
+  'did:peer:',
+  'did:pistis:',
+  'did:pkh:',
+  'did:polygon:',
+  'did:ptn:',
+  'did:safe:',
+  'did:san:',
+  'did:schema:',
+  'did:selfkey:',
+  'did:signor:',
+  'did:sirius:',
+  'did:sol:',
+  'did:sov:',
+  'did:ssb:',
+  'did:ssw:',
+  'did:stack:',
+  'did:tangle:',
+  'did:tls:',
+  'did:trust:',
+  'did:trustbloc:',
+  'did:trx:',
+  'did:ttm:',
+  'did:tyron:',
+  'did:twit:',
+  'did:tys:',
+  'did:tz:',
+  'did:unik:',
+  'did:unisot:',
+  'did:uns:',
+  'did:uport:',
+  'did:v1:',
+  'did:vaa:',
+  'did:vaultie:',
+  'did:vid:',
+  'did:vivid:',
+  'did:vvo:',
+  'did:web:',
+  'did:wlk:',
+  'did:work:',
+  'did:lac:',
+  'did:pml:',
+];
+
+export function assertValidMetadata(
+  opMetadata: DiscoveryMetadataPayload,
+  rpMetadata: RPRegistrationMetadataPayload
+): SupportedTypes {
+  let methods = [];
+  const credentials = verifyCredentialsFormat(
+    rpMetadata.credential_formats_supported,
+    opMetadata.credential_formats_supported
+  );
   const isDid = verifySubjectIdentifiers(rpMetadata.subject_identifiers_supported);
   if (isDid && rpMetadata.did_methods_supported) {
-    verifyDidMethods(rpMetadata.did_methods_supported, opMetadata.did_methods_supported);
+    methods = verifyDidMethods(rpMetadata.did_methods_supported, opMetadata.did_methods_supported);
+  } else if (isDid && (!rpMetadata.did_methods_supported || !rpMetadata.did_methods_supported.length)) {
+    methods = verifyDidMethods(allDidMethodsSupported, opMetadata.did_methods_supported);
```

Comment:
Is the intent here that whenever the RP does not expose which DIDs it supports to accept everything, but then only the ones in our internal list above? If so we should just leave it up to the RP to try to resolve the actual DID that the OP provides. If that doesn't work, the RP will return an error anyway. Right now we are restricting to our internal list (see the 'shinynew' remark from earlier

Reply:
Fixed!